diff --git a/dlls/mshtml/htmldoc.c b/dlls/mshtml/htmldoc.c
index 72f5f68..583e34c 100644
--- a/dlls/mshtml/htmldoc.c
+++ b/dlls/mshtml/htmldoc.c
@@ -1566,16 +1566,40 @@ static HRESULT WINAPI HTMLDocument_queryCommandSupported(IHTMLDocument2 *iface,
         VARIANT_BOOL *pfRet)
 {
     HTMLDocumentNode *This = impl_from_IHTMLDocument2(iface);
-    FIXME("(%p)->(%s %p)\n", This, debugstr_w(cmdID), pfRet);
-    return E_NOTIMPL;
+    OLECMDID cmdid;
+
+    TRACE("(%p)->(%s %p)\n", This, debugstr_w(cmdID), pfRet);
+
+    if(!pfRet)
+        return E_POINTER;
+
+    /* A command is supported if it is a known command identifier. */
+    *pfRet = variant_bool(cmdid_from_string(cmdID, &cmdid));
+    return S_OK;
 }
 
 static HRESULT WINAPI HTMLDocument_queryCommandEnabled(IHTMLDocument2 *iface, BSTR cmdID,
         VARIANT_BOOL *pfRet)
 {
     HTMLDocumentNode *This = impl_from_IHTMLDocument2(iface);
-    FIXME("(%p)->(%s %p)\n", This, debugstr_w(cmdID), pfRet);
-    return E_NOTIMPL;
+    OLECMD cmd = {0};
+    OLECMDID cmdid;
+    HRESULT hres;
+
+    TRACE("(%p)->(%s %p)\n", This, debugstr_w(cmdID), pfRet);
+
+    if(!pfRet)
+        return E_POINTER;
+
+    *pfRet = VARIANT_FALSE;
+    if(!cmdid_from_string(cmdID, &cmdid))
+        return OLECMDERR_E_NOTSUPPORTED;
+
+    cmd.cmdID = cmdid;
+    hres = IOleCommandTarget_QueryStatus(&This->IOleCommandTarget_iface, &CGID_MSHTML, 1, &cmd, NULL);
+    if(SUCCEEDED(hres) && (cmd.cmdf & OLECMDF_ENABLED))
+        *pfRet = VARIANT_TRUE;
+    return S_OK;
 }
 
 static HRESULT WINAPI HTMLDocument_queryCommandState(IHTMLDocument2 *iface, BSTR cmdID,
