diff --git a/dlls/ntdll/version.c b/dlls/ntdll/version.c
index c58fa2f..7a03d5b 100644
--- a/dlls/ntdll/version.c
+++ b/dlls/ntdll/version.c
@@ -172,7 +172,7 @@ static const RTL_OSVERSIONINFOEXW VersionData[NB_WINDOWS_VERSIONS] =
     },
     /* WIN11 */
     {
-        sizeof(RTL_OSVERSIONINFOEXW), 10, 0, 22000, VER_PLATFORM_WIN32_NT,
+        sizeof(RTL_OSVERSIONINFOEXW), 10, 0, 26100, VER_PLATFORM_WIN32_NT,
         L"", 0, 0, VER_SUITE_SINGLEUSERTS, VER_NT_WORKSTATION, 0
     },
 };
diff --git a/programs/winecfg/appdefaults.c b/programs/winecfg/appdefaults.c
index 586b72a..12dd7c4 100644
--- a/programs/winecfg/appdefaults.c
+++ b/programs/winecfg/appdefaults.c
@@ -50,7 +50,7 @@ struct win_version
 
 static const struct win_version win_versions[] =
 {
-    { L"win11",     L"Windows 11",      L"6.3", 10,  0, 22000,  588, VER_PLATFORM_WIN32_NT, L"", 0, 0, L"WinNT"},
+    { L"win11",     L"Windows 11",      L"6.3", 10,  0, 26100,  588, VER_PLATFORM_WIN32_NT, L"", 0, 0, L"WinNT"},
     { L"win10",     L"Windows 10",      L"6.3", 10,  0, 19045, 5796, VER_PLATFORM_WIN32_NT, L"", 0, 0, L"WinNT"},
     { L"win81",     L"Windows 8.1",     NULL,    6,  3,  9600,    0, VER_PLATFORM_WIN32_NT, L"", 0, 0, L"WinNT"},
     { L"win8",      L"Windows 8",       NULL,    6,  2,  9200,    0, VER_PLATFORM_WIN32_NT, L"", 0, 0, L"WinNT"},
